tests/test_instruments_qm.py

```diff
@@ -2,8 +2,10 @@
 
 import numpy as np
 import pytest
-from qm import qua
 
+qua = pytest.importorskip("qm.qua")
```

Comment:
Currently these tests are skipped anyway in the conftest. I guess when we re-enable them we will have to manually skip them for 3.12, otherwise they would fail? This should still be fine. Unless we are lucky and qm-qua adds support for 3.12 at that time, in which case we can just lift the constraints.

Reply:
Despite the tests being individually skipped, they were still loaded, and this was making this import failing for py3.12.

This function is skipping this entire module if the dependency is not available. But we could do the same in the `conftest.py`: associate to the platform the list of dependencies, and attempt to load (or just attempt to load the platform). If it fails for a `ModuleNotFoundError`, we just skip the test.

I would be in favor of using `pytest.importorskip` even for the other drivers, since they are optional. At least, we could advance our support independently of each driver (while, if NumPy or Qibo has no support for some system, there is little point in having Qibolab supporting it).

Reply:
> Despite the tests being individually skipped, they were still loaded, and this was making this import failing for py3.12.
>
> This function is skipping this entire module if the dependency is not available. But we could do the same in the `conftest.py`: associate to the platform the list of dependencies, and attempt to load (or just attempt to load the platform). If it fails for a `ModuleNotFoundError`, we just skip the test.

Yes, I knew the first part, that the import would cause issues even if we skip, but I didn't know the second, that the tests would be skipped if the dependency is not available. I would say this is exactly what we want.

> I would be in favor of using `pytest.importorskip` even for the other drivers, since they are optional. At least, we could advance our support independently of each driver (while, if NumPy or Qibo has no support for some system, there is little point in having Qibolab supporting it).

Sounds good to me, too.

Reply:
Now in #997